packages/react/src/tools/wrapComponent.js

```diff
@@ -10,7 +10,16 @@ import PropTypes from 'prop-types';
 import cx from 'classnames';
 import { usePrefix } from '../internal/usePrefix';
 
+/**
+ * @param {{ name: string, className?: string, type: string }} param0
```

Comment:
```suggestion
 * @param {{ name: string, className?: string | Function, type: string }} param0
```

Reply:
Ah ty I missed the function usage. I would prefer something more strict than `Function` though. Would you settle for this instead? 

```
@param {{ name: string, type: string, className?: string | (prefix: string) => string }} props
```

Reply:
sounds good! though technically the `getClassName` function could return an object with classNames as keys and it'd still work 🤔: 
<img width="823" alt="image" src="https://user-images.githubusercontent.com/40550942/223276185-0b39887b-5c5b-44b2-9a0d-e5f50a5b325c.png">

Reply:
Ah right because most components consume the className with that `classnames` lib. I didnt see any current usages in the code like that, I could change it to this. might be a bit too wordy for a jsdoc typing

<img width="667" alt="image" src="https://user-images.githubusercontent.com/99205866/223515893-bcfbf20e-5016-4206-8453-6de2bacb0b66.png">